okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -151,6 +151,31 @@ public void onResponse(Call _call, Response response) {
         }
     }
 
+    private InterruptedIOException handleInterruption(SettableFuture<Response> future) {
+        getDelegate().cancel();
+        // Regardless of the cancel above, the call may have succeeded or is going to succeed, and we need to close the
+        // response body correctly:
+        // (1) if the success callback gets called before we try to set the exception, we must close the response
+        // (2) if the success callback gets called after we set the exception, the callback will close the response
+        if (!future.setException(new InterruptedIOException("Call cancelled via interruption"))) {
+            try {
+                Response response = Futures.getDone(future);
+                closeResponseBody(response);
```

Comment:
I think we could make this a lot simpler by just adding a listener to the future that will close the response body. This will ensure it still gets closed even if we are interrupted while waiting.

Reply:
I *think* this is equivalent, going off of `ListenableFuture#addListener` javadoc which `Futures#addCallback` delegates to eventually. Even if we have to use the `directExecutor`, but that's also equivalent to this change too I believe.

IIUC this is the reasoning:
1) If we interrupt before the request completes, calling thread is no longer blocking on `Future#get`, we then `Futures#addCallback` in the catch block, and then the thread that will eventually complete the request will then close the body.

2) If we interrupt after the request completes, calling thread is no longer blocking on `Future#get`, and adding the same callback would then cause the response to be closed immediately but in the calling thread

We would also stop trying to close the response body in the normal callback in lines 122 right?

Reply:
Makes sense, this is much cleaner.